proof/archive.go

```diff
@@ -57,6 +58,21 @@ type Locator struct {
 	ScriptKey btcec.PublicKey
 }
 
+// Hash returns a SHA256 hash of the bytes serialized locator.
+func (l *Locator) Hash() [32]byte {
+	var buf bytes.Buffer
+	if l.AssetID != nil {
```

Comment:
silly question can the `AssetID` be nil?

Reply:
According to the locator struct:
```
// AssetID the asset ID of the proof to fetch. This is an optional field.
AssetID *asset.ID
```

which means that `AssetID` can technically be `nil`.

Here's an example in code where `AssetID` is `nil`: https://github.com/lightninglabs/taro/blob/e2485efba4daf4cd87889269b16f3c6ba7e7606d/proof/archive.go#L288-L292